lib/memory.js

```diff
@@ -18,37 +18,51 @@ const UNLIMITED_MEMORY_AMOUNT = 9223372036854771712;
 
 /**
  * Reads metrics from `/sys/fs/cgroup/memory/memory.stat` and `/sys/fs/cgroup/memory/memory.kmem.usage_in_bytes`
- * @returns {Number} container usage based on rss and kmemUsage stats
+ * @returns {Number} container usage based on rss and kmemUsage stats (bytes)
  */
 function containerUsage() {
-	const rss = readandFormatMetric('memory/memory.stat');
-	const kmemUsage = readandFormatMetric('memory/memory.kmem.usage_in_bytes');
-	if (typeof(rss) !== "number" || typeof(kmemUsage) !== "number" || (isNaN(rss)) || (isNaN(kmemUsage)) ) {
-		throw Error(`One or more metrics are malformed. rss: ${rss}, kmemUsage: ${kmemUsage}`);
-	}
-	return (rss + kmemUsage);
+    const rss = readandFormatMetric('memory/memory.stat');
+    const kmemUsage = readandFormatMetric('memory/memory.kmem.usage_in_bytes');
+    if (typeof (rss) !== "number" || typeof (kmemUsage) !== "number" || (isNaN(rss)) || (isNaN(kmemUsage))) {
+        throw Error(`One or more metrics are malformed. rss: ${rss}, kmemUsage: ${kmemUsage}`);
+    }
+    return (rss + kmemUsage);
 }
 
 /**
  * Reads metrics from `/sys/fs/cgroup/memory`
  * @returns {Number} container usage divided by memory limit of the container
  */
-function containerUsagePercentage(memoryContainerUsage=false){
-	if (!(memoryContainerUsage)) {
-		memoryContainerUsage = containerUsage();
-	}
-	let limit = readandFormatMetric('memory/memory.limit_in_bytes');
-	if (typeof(memoryContainerUsage) !== "number" || typeof(limit) !== "number" || (isNaN(memoryContainerUsage)) || (isNaN(limit)) ) {
-		throw Error(`One or more metrics are malformed. containerUsage: ${memoryContainerUsage}, limit: ${limit}`);
-	}
-	// if it's set to unlimited, return the total memory of the host
-	if (limit >= UNLIMITED_MEMORY_AMOUNT || limit === 0) {
-		limit = os.totalmem();
-	}
-	return (memoryContainerUsage / limit) * 100;
+function containerUsagePercentage(memoryContainerUsage = false) {
+    if (!(memoryContainerUsage)) {
+        memoryContainerUsage = containerUsage();
+    }
+    const limit = containerMemoryLimit(memoryContainerUsage);
+    return (memoryContainerUsage / limit) * 100;
+}
+
+/**
+ * Reads metrics from `/sys/fs/cgroup/memory`
+ * @returns {Number} container memory limit (bytes)
+ */
+function containerMemoryLimit(memoryContainerUsage = false) {
+    if (!(memoryContainerUsage)) {
```

Comment:
I don't think this block or the memoryContainerUsage parameter are needed here

Reply:
Correct again!